File: goal-app-frontend/src/types/goal.ts
export interface Goal {
  id: number;
  title: string;
  description: string;
  createdBy: number;
  type: 'PERSONAL' | 'SHARED';
  difficulty: 'EASY' | 'MEDIUM' | 'HARD';
  deadline: string;
  createdAt: string;
  assignments: GoalAssignment[];
}

export interface GoalAssignment {
  userId: number;
  status: 'PENDING' | 'COMPLETED' | 'MISSED';
  completedAt?: string;
  lastUpdated: string;
}

export interface CreateGoalData {
  title: string;
  description: string;
  createdBy: number;
  type: 'PERSONAL' | 'SHARED';
  difficulty: 'EASY' | 'MEDIUM' | 'HARD';
  deadline: string;
  assignedUserIds?: number[];
}

File: goal-app-backend/goal-service/src/main/java/org/goalapp/goal/entities/Goal.java
package org.goalapp.goal.entities;
import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "goals")
public class Goal {

    public LocalDateTime getDeadline() {
        return deadline;
    }

    public void setDeadline(LocalDateTime deadline) {
        this.deadline = deadline;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public Long getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(Long createdBy) {
        this.createdBy = createdBy;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Difficulty getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(Difficulty difficulty) {
        this.difficulty = difficulty;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public GoalType getType() {
        return type;
    }

    public void setType(GoalType type) {
        this.type = type;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;

    private Long createdBy;

    @Enumerated(EnumType.STRING)
    private GoalType type = GoalType.PERSONAL;

    @Enumerated(EnumType.STRING)
    private Difficulty difficulty = Difficulty.MEDIUM;

    private LocalDateTime deadline;
    private LocalDateTime createdAt = LocalDateTime.now();

    public enum GoalType { PERSONAL, SHARED }
    public enum Difficulty { EASY, MEDIUM, HARD }
}